src/db.js

```diff
@@ -0,0 +1,43 @@
+import Dexie from 'dexie'
+
+const db = new Dexie('EM')
+
+const initHelpers = async () => {
+  const staticHelpersExist = await db.helpers.get({ id: 'EM' })
+  if (!staticHelpersExist) {
+    await db.helpers.add({ id: 'EM' })
+  }
+}
+
+const initDB = async () => {
+  await db.version(1).stores({
+    thoughtIndex: 'id, value, *contexts, created, lastUpdated',
+    contextIndex: 'id, *context, lastUpdated',
+    helpers: 'id, cursor, lastUpdated, recentlyEdited, schemaVersion'
+  })
+  await initHelpers()
+}
+
+export const updateThoughtIndex = async (id, thought) => db.thoughtIndex.put({ id, ...thought })
+export const bulkUpdateThoughtIndex = async thoughtIndexMap => {
+  const thoughtsArray = Object.keys(thoughtIndexMap).map(key => ({ ...thoughtIndexMap[key], id: key }))
+  return db.thoughtIndex.bulkPut(thoughtsArray)
+}
```

Comment:
Great, this should be more efficient!

Let's rename:
- `getThoughtIndexById` → `getThought`
- `updateThoughtIndex` → `updateThought`
- `bulkUpdateThoughtIndex` → `updateThoughtIndex`
- `updateContextIndex` → `updateContext`
- `bulkUpdateContextIndex` → `updateContextIndex`
- `deleteThoughtIndex` → `deleteThought`
- `deleteContextIndex` → `deleteContext`
- `getContextIndexes` → `getContextIndex`
- `getThoughtIndexes` → `getThoughtIndex`

Within `getThoughtIndexes` and `getContextIndexes` use [`_.keyBy`](https://lodash.com/docs/4.17.15#keyBy) to convert the array to an object within the getter.

Reply:
_.keyBy doesn't seem to work with `contextIndex` due to its structure. So I've used `reduce`

Reply:
Interesting. Okay.